src/ga4gh/vrs/models.py

```diff
@@ -431,6 +456,29 @@ class SequenceLocation(_Ga4ghIdentifiableObject):
     )
     sequence: Optional[SequenceString] = Field(None, description="The literal sequence encoded by the `sequenceReference` at these coordinates.")
 
+    def ga4gh_serialize_as_version(self, as_version):
+        """This method will return a serialized string following the conventions for
+        SequenceLocation serialization as defined in the VRS version specified by 'as_version`."""
+        if as_version == '1.3':
+            out = list()
+            for value in [self.start,self.end]:
+                value = get_pydantic_root(value)
+                if isinstance(value, int):
+                    result = f'{{"type":"Number","value":{value}}}'
+                elif isinstance(value, list):
+                    if value[0] is None:
+                        result = f'{{"comparator":"<=","type":"IndefiniteRange","value":{value[1]}}}'
+                    elif value[1] is None:
+                        result = f'{{"comparator":">=","type":"IndefiniteRange","value":{value[0]}}}'
+                    else:
+                        result = f'{{"max":{value[1]},"min":{value[0]},"type":"DefiniteRange"}}'
+                else:
+                    raise ValueError(f'{value} is not int or list.')
+                out.append(result)
+            return f'{{"interval":{{"end":{out[1]},"start":{out[0]},"type":"SequenceInterval"}},"sequence_id":"{self.sequenceReference.refgetAccession.split('.')[1]}","type":"SequenceLocation"}}'
```

Comment:
This assumes that `sequenceReference` is not null *and* that it is type `SequenceReference`. We should handle cases where `sequenceReference` is null and cases where `sequenceReference` is an `IRI`.

Reply:
Yes, agreed.